Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide.hs

```diff
@@ -227,84 +233,98 @@ endRide handle@ServiceHandle {..} rideId req = withLogTag ("rideId-" <> rideId.g
 
-  tripEndPoint <- case req of
+  (tripEndPoint, odometerEndReading) <- case req of
     DriverReq driverReq -> do
       logTagInfo "driver -> endRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId rideOld.id)
-      pure driverReq.point
+      pure (driverReq.point, driverReq.odometerEndReading :: Maybe Int)
     DashboardReq dashboardReq -> do
       logTagInfo "dashboard -> endRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId rideOld.id)
       case dashboardReq.point of
-        Just point -> pure point
+        Just point -> pure (point, Nothing)
         Nothing -> do
-          pure $ getCoordinates booking.toLocation
+          pure (getCoordinates booking.toLocation, Nothing)
     CronJobReq cronJobReq -> do
       logTagInfo "cron job -> endRide : " ("DriverId " <> getId driverId <> ", RideId " <> getId rideOld.id)
       case cronJobReq.point of
-        Just point -> pure point
+        Just point -> pure (point, Nothing)
         Nothing -> do
-          pure $ getCoordinates booking.toLocation
+          pure (getCoordinates booking.toLocation, Nothing)
     CallBasedReq _ -> do
-      pure $ getCoordinates booking.toLocation
+      pure (getCoordinates booking.toLocation, Nothing)
 
   goHomeConfig <- CQGHC.findByMerchantId booking.providerId
   ghInfo <- CQDGR.getDriverGoHomeRequestInfo driverId booking.providerId (Just goHomeConfig)
 
   homeLocationReached' <-
-    if ghInfo.status == Just DDGR.ACTIVE && goHomeConfig.enableGoHome
-      then do
-        case ghInfo.driverGoHomeRequestId of
-          Nothing -> do
-            logError "DriverGoHomeRequestId not present even though status is active."
-            return Nothing
-          Just ghrId -> do
-            mbDriverGoHomeReq <- QDGR.findById ghrId
-            case mbDriverGoHomeReq of
-              Just driverGoHomeReq -> do
-                let driverHomeLocation = Maps.LatLong {lat = driverGoHomeReq.lat, lon = driverGoHomeReq.lon}
-                routesResp <- DMaps.getTripRoutes (driverId, booking.providerId) (buildRoutesReq tripEndPoint driverHomeLocation)
-                logDebug $ "Routes resp for EndRide API :" <> show routesResp <> "(source, dest) :" <> show (tripEndPoint, driverHomeLocation)
-                let driverHomeDists = mapMaybe (.distance) routesResp
-                if any ((<= goHomeConfig.destRadiusMeters) . getMeters) driverHomeDists
-                  then do
-                    CQDGR.deactivateDriverGoHomeRequest booking.providerId driverId DDGR.SUCCESS ghInfo (Just True)
-                    return $ Just True
-                  else do
-                    CQDGR.resetDriverGoHomeRequest booking.providerId driverId goHomeConfig ghInfo
-                    return $ Just False
-              Nothing -> return Nothing
-      else return Nothing
+    case odometerEndReading of
+      Nothing -> do
+        if ghInfo.status == Just DDGR.ACTIVE && goHomeConfig.enableGoHome
```

Comment:
instead of having case match in odometerEndReading just add && condition here if `isNothing odometerEndReading`

Reply:
Yes, already have mentioned this
<img width="447" alt="Screenshot 2023-10-17 at 12 16 02 PM" src="https://github.com/nammayatri/nammayatri/assets/33424495/4be42937-f7de-43a6-bdb8-ccc5f31e727b">